File: services/go-gin/storage.go
package main

// LogStorage определяет интерфейс для работы с логом
type LogStorage interface {
	WriteLine(line string) error
	ReadAll() (string, error)
}

// TimeFormatter определяет интерфейс для форматирования времени
type TimeFormatter interface {
	Format() string
}

File: services/rust-actix/src/storage.rs
use std::io;

/// Интерфейс для работы с хранилищем логов (SRP, DIP)
pub trait LogStorage: Send + Sync {
    fn write_line(&self, line: &str) -> Result<(), io::Error>;
    fn read_all(&self) -> Result<String, io::Error>;
}
